### PR TITLE
Implement File Service Support Infrastructure

### LINKED ISSUE #1000: Multi-Cloud File Service Framework Support
**Issue/Feature Description:** Add Support new File service into multi-cloud

**Why this issue to fixed / feature is needed(give scenarios or use cases):* New service for supporting file type backend is required

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by himanshuvar**
**What type of PR is this?**
> /kind new feature

**What this PR does / why we need it**: This PR is to implement File Service Support Infrastructure which includes

- Add File Service Docker configuration
- Add Base Framework for File Support
- Update docker-compose.yml

**Which issue(s) this PR fixes**: 

Fixes #1000 

**Test Report Added?**:

> /kind TESTED

**Test Report**:

**Running Docker Services:**
root@ubuntu:~/gopath/src/github.com/opensds/multi-cloud# docker ps -a
CONTAINER ID        IMAGE                                  COMMAND                  CREATED             STATUS              PORTS                                                NAMES
100c21f59322        opensdsio/multi-cloud-dataflow         "/dataflow"              8 minutes ago       Up 8 minutes                                                             multi-cloud_dataflow_1_3219c2f76e16
592de72dce55        opensdsio/multi-cloud-datamover        "/datamover"             8 minutes ago       Up 8 minutes                                                             multi-cloud_datamover_1_f3e89d50dcd8
c1c2db67e4b6        opensdsio/multi-cloud-s3               "/initdb.sh"             8 minutes ago       Up 8 minutes                                                             multi-cloud_s3_1_2b4921c29f55
6a201206998a        wurstmeister/kafka:2.11-2.0.1          "start-kafka.sh"         8 minutes ago       Up 8 minutes        0.0.0.0:9092->9092/tcp                               multi-cloud_kafka_1_e080ed6c2ea0
0fc57eab9994        wurstmeister/zookeeper                 "/bin/sh -c '/usr/sb…"   8 minutes ago       Up 8 minutes        22/tcp, 2888/tcp, 3888/tcp, 0.0.0.0:2181->2181/tcp   multi-cloud_zookeeper_1_36a5d513f288
d577863180ab        opensdsio/multi-cloud-api              "/api"                   8 minutes ago       Up 8 minutes        0.0.0.0:8090->8090/tcp                               multi-cloud_s3api_1_3a7850f19900
1ffbbdd2d8c6        redis                                  "docker-entrypoint.s…"   8 minutes ago       Up 8 minutes        0.0.0.0:6379->6379/tcp                               redis
20b6041c987d        opensdsio/multi-cloud-file             "/file"                  8 minutes ago       Up 8 minutes                                                             multi-cloud_file_1_785acc431f68
769f6a7036cd        opensdsio/multi-cloud-api              "/api"                   8 minutes ago       Up 8 minutes        0.0.0.0:8089->8089/tcp                               multi-cloud_api_1_cdaecc64eeeb
cc74f5b8823b        mongo                                  "docker-entrypoint.s…"   8 minutes ago       Up 8 minutes        0.0.0.0:27017->27017/tcp                             multi-cloud_datastore_1_639a5c325040
e55e1c42bbb0        opensdsio/multi-cloud-backend          "/backend"               8 minutes ago       Up 8 minutes                                                             multi-cloud_backend_1_33faa0c3bab5
eca8d36b1cb9        pingcap/tidb:v2.1.16                   "/tidb-server --stor…"   8 minutes ago       Up 8 minutes        0.0.0.0:4000->4000/tcp, 0.0.0.0:10080->10080/tcp     tidb
5444da1043ba        opensdsio/dashboard:v0.10.0            "/bin/sh -c /opt/das…"   6 weeks ago         Up 6 weeks                                                               dashboard
e86f14e19337        opensdsio/opensds-authchecker:latest   "/sbin/init"             6 weeks ago         Up 6 weeks                                                               opensds-authchecker

**File Service Docker Logs:**
`root@ubuntu:~/gopath/src/github.com/opensds/multi-cloud# docker logs -f 20b6041c987d
[2020-06-09T15:59:22+05:30] [INFO] [/root/gopath/src/github.com/opensds/multi-cloud/file/cmd/main.go] [main.main():35] [PID:1] Initializing File service..
[2020-06-09T15:59:22+05:30] [INFO] [/root/gopath/src/github.com/opensds/multi-cloud/file/pkg/service/service.go] [service.NewFileService():26] [PID:1] Init file service finished.

2020-06-09 15:59:22  file=server.go:185 level=error rpc Register: type File has no exported methods of suitable type
2020-06-09 15:59:22  file=service.go:206 level=info Starting [service] file
2020-06-09 15:59:22  file=grpc.go:845 level=info Server [grpc] Listening on [::]:42458
2020-06-09 15:59:22  file=grpc.go:676 level=info Registry [mdns] Registering node: file-878eedcf-2778-484d-82f6-f4e00f9e5580`

**Special notes for your reviewer**:
This PR is just a blueprint for file service.

### CODE REVIEW

File: file/cmd/main_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package main
```
Comment (kumarashit) [inline file/cmd/main_test.go:1]: Should the copyright be updated here?

Comment (himanshuvar) [reply file/cmd/main_test.go:15]: Yes, I have updated the copyright. 
Thank You

File: docker-compose.yml [hunk 2]
```diff
@@ -80,6 +80,16 @@ services:
     ports:
       - "6379:6379"
 
+  file:
+    image: opensdsio/multi-cloud-file
+    environment:
+      - MICRO_REGISTRY=mdns
+      - SVC_FLAG=file
```
Comment (sfzeng) [inline docker-compose.yml:87]: SVC_FLAG may not required here.

File: Makefile [hunk 3]
```diff
@@ -54,6 +57,10 @@ docker: build
 	chmod 755 backend/backend
 	docker build backend -t opensdsio/multi-cloud-backend:latest
 
+	cp $(BUILD_DIR)/file file
+	chmod 755 file/file
+	docker build file -t opensdsio/multi-cloud-file:latest
+
```
Comment (sfzeng) [inline Makefile:63]: File may also need to be cleaned, onle line 81.

Comment (himanshuvar) [reply Makefile:63]: okay. Done.
Thanks

File: docker-compose.yml (hunk 2 above)
Comment (himanshuvar) [reply docker-compose.yml]: okay. Done.
Thanks

**Pull request merged**